src/mame/drivers/firebeat.cpp

```diff
@@ -159,9 +162,208 @@
 #include "speaker.h"
 #include "osdcomm.h"
 
+#include "wdlfft/fft.h"
+
 #include "firebeat.lh"
 
 
+/*****************************************************************************/
+DECLARE_DEVICE_TYPE(KONAMI_FIREBEAT_EXTEND_SPECTRUM_ANALYZER, firebeat_extend_spectrum_analyzer_device)
```

Comment:
I debated whether this deserved to be its own machine in `src/mame/machines` or not. This particular code is only ever used for beatmania III, and the reasoning for making it a separate device is because I wanted to make it a `device_mixer_interface` for audio processing.